### PR TITLE
#476 P2P Side-Channel Broadcast

### PR DESCRIPTION
**Pull request by wolfposd**
Related to https://github.com/tendermint/tendermint/issues/476

- Created a low priority reactor "TransientBroadcast"
- Take TransientTx from RPC/WSRPC
- Broadcast to Peers
- Re-Broadcast to own subscribed RPC-Clients via EventSwitch

### CODE REVIEW

File: p2p/transient_broadcast_reactor.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
+package p2p
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/tendermint/go-wire"
+	"github.com/tendermint/tendermint/types"
+	"reflect"
```
Comment (zramsay) [inline p2p/transient_broadcast_reactor.go]: should be below `"fmt"` then line break before the tendermint imports

**Comment by wolfposd** [comment]
this is still not complete yet, might have issued this too early.

- gossiping needs to be adressed
- not all peers that are connect receive a message, peer-list=1 even with two connected nodes

**Comment by adrianbrink** [comment]
@wolfposd Is this something that might behave similar to Whisper for Ethereum?

**Comment by wolfposd** [comment]
Yes, Whisper is describing our use-cases for this feature pretty good

> DApps that need to signal to each other in order to ultimately collaborate on a transaction. For example, a currency exchange DApp may use it to coordinate an offer prior to creating one (or two, depending on how the exchange is structured) transactions on the exchange.

**Comment by wolfposd** [comment]
requires https://github.com/tendermint/tmlibs/pull/63/commits

**Comment by wolfposd** [comment]
updated to newest develop, with eventbus

**Comment by zramsay** [comment]
rebase? do we still want this?

**Comment by ebuchman** [comment]
Really appreciate this PR, but unfortunately it would be too much of a distraction to try to get this merged in the short term while we're working towards cosmos launch. we'd need some discussion around design, perhaps ideally by starting with a reactor spec in the same vein as the others under https://github.com/tendermint/tendermint/tree/develop/docs/specification/new-spec/reactors. Of course, it would also need some tests.

I'll close it for now, and hopefully we can revisit later in the year.

**Pull request closed without merging**